### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by nibra**
### Major refactoring

* Remove avoidable dependencies
* Remove code duplication
* Use methods instead of comments
* Joomla 4 compatibility
* Remove calls to deprecated code

### CODE REVIEW

File: plg_cccsocialmedia/cccsocialmedia.php [hunk 1]
```diff
@@ -48,500 +60,576 @@ class plgSystemCccsocialmedia extends CMSPlugin
...
-	function onContentPrepareForm($form, $data)
+	/**
+	 * Add fields for the OpenGraph data to the form
+	 *
+	 * @param   EventInterface  $event  The event
+	 *
+	 * @return boolean
+	 * @since  1.0
+	 */
+	public function onContentPrepareForm(EventInterface $event): bool
```
Comment (coolcat-creations) [inline plg_cccsocialmedia/cccsocialmedia.php]: Something is wrong here and gives an Error, is the EventInterface right? Shouldn't it be Form $form or something like that?

File: plg_cccsocialmedia/cccsocialmedia.php [hunk 2]
```diff
@@ -48,500 +60,576 @@ class plgSystemCccsocialmedia extends CMSPlugin
...
+	/**
+	 * Prepare the OpenGraph metadata for rendeering
+	 *
+	 * @return bool
+	 * @since  1.0
+	 */
+	public function onBeforeCompileHead(): bool
 	{
-		$document = Factory::getDocument();
-		$config = Factory::getConfig();
```
Comment (coolcat-creations) [inline plg_cccsocialmedia/cccsocialmedia.php:123]: seems to be missing as an error shows up in the frontend Call to undefined method Joomla\CMS\Application\SiteApplication::getConfig()

**Comment by coolcat-creations** [comment]
Unfortunately I have still a Call to undefined method Joomla\CMS\Application\SiteApplication::getConfig() Error in J3...

File: plg_cccsocialmedia/cccsocialmedia.php [hunk 3]
```diff
@@ -9,10 +9,14 @@
...
 use Joomla\CMS\Application\CMSApplication;
-use Joomla\CMS\Factory;
+use Joomla\CMS\Document\Document;
+use Joomla\CMS\Form\Form;
 use Joomla\CMS\Plugin\CMSPlugin;
+use Joomla\CMS\Table\Content;
 use Joomla\CMS\Uri\Uri;
 use Joomla\Database\DatabaseDriver;
+use Joomla\Event\EventInterface;
+use Joomla\Registry\Registry;
```
Comment (coolcat-creations) [inline plg_cccsocialmedia/cccsocialmedia.php:20]:
```suggestion
use Joomla\CMS\Factory;

```

File: plg_cccsocialmedia/cccsocialmedia.php [hunk 4]
```diff
@@ -48,500 +61,547 @@ class plgSystemCccsocialmedia extends CMSPlugin
...
+			'tw_title',
+			'tw_description',
+			'tw_image',
+			'tw_image_alt',
+			'tw_site',
+			'tw_type',
+			'tw_creator'
+		);
+
+		$params->set('sitename', $this->app->getConfig()->get('sitename'));
```
Comment (coolcat-creations) [inline plg_cccsocialmedia/cccsocialmedia.php]:
```suggestion
		$params->set('sitename', Factory::getApplication()->get('sitename'));
```

**Review by coolcat-creations — CHANGES_REQUESTED** [review]
@nibra please check my suggestions, this should work then in J3 too (see my error from my last comment)

**Comment by coolcat-creations** [comment]
@nibra please check my suggestions, this should work then in J3 too (see my error from my last comment)

File: plg_cccsocialmedia/cccsocialmedia.php (hunk 1 above)
Comment (nibra) [reply plg_cccsocialmedia/cccsocialmedia.php]: I already changed that back to form with the latest commit...

Comment (coolcat-creations) [reply plg_cccsocialmedia/cccsocialmedia.php]: Yes that's an old comment :) I meant my other one

**Comment by nibra** [comment]
Try this- I currently have no J3 running to test it myself.

**Pull request merged**